addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/CsrfTokenScanRule.java

```diff
@@ -123,7 +123,8 @@ public void init() {
     @Override
     public void scan() {
         if ((AlertThreshold.HIGH.equals(getAlertThreshold()) && !getBaseMsg().isInScope())
-                || !getBaseMsg().getResponseHeader().isHtml()) {
+                || !getBaseMsg().getResponseHeader().isHtml()
+                || isPage404(getBaseMsg())) {
```

Comment:
I'm okay with this.

Reply:
I don't think we should just ignore based on status code, this should be documented in the help page anyway.

Reply:
I was assuming the argument here is something like: If the original response was 404 (or 404 as determined by Analyser/Custom Pages) then what form is there that we care about…likely just a search form with no token….

But I’m fully open to debate it more.

The following change was much more concerning to me, assuming that only 200 is success.